thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js

```diff
@@ -530,6 +598,79 @@ export default Ember.Controller.extend({
     this._checkSession();
   },
 
+  //
+  // default selection
+  //
+
+  /**
+   * Transition to route for metric id
+   *
+   * @private
+   */
+  _setupContext() {
+    const { context } =
+      this.getProperties('context');
+
+    const contextMetricUrns = filterPrefix(context.urns, 'thirdeye:metric:');
+
+    if (_.isEmpty(contextMetricUrns)) { return; }
+
+    const metricUrn = contextMetricUrns[0];
+    const metricId = metricUrn.split(':')[2];
+
+    this.transitionToRoute({ queryParams: { metricId } });
+  },
+
+  /**
+   * Select top-scoring entities by default if metric selected for the first time.
+   * Idempotent addition of urns to support multiple execution while data loading.
+   *
+   * @private
+   */
+  _setupSelected() {
+    const { context, entities, scores, selectedUrns, loadingFrameworks, isLoadingScores } =
+      this.getProperties('context', 'entities', 'scores', 'selectedUrns', 'loadingFrameworks', 'isLoadingScores');
+
+    const newSelectedUrns = new Set(selectedUrns);
+
+    const contextMetricUrns = filterPrefix(context.urns, 'thirdeye:metric:');
+    contextMetricUrns.forEach(urn => newSelectedUrns.add(urn));
+    contextMetricUrns.map(toCurrentUrn).forEach(urn => newSelectedUrns.add(urn));
+    contextMetricUrns.map(toBaselineUrn).forEach(urn => newSelectedUrns.add(urn));
+
+    // events
+    const groupedEvents = Object.values(entities)
+      .filter(e => e.type === 'event')
+      .reduce((agg, e) => {
+        const type = e.eventType;
+        agg[type] = agg[type] || [];
+        agg[type].push(e);
+        return agg;
+      }, {});
+
+    Object.values(groupedEvents)
+      .forEach(arr => {
+        const topk = arr.filter(e => e.score >= ROOTCAUSE_SETUP_EVENTS_SCORE_THRESHOLD);
+        topk.forEach(e => newSelectedUrns.add(e.urn));
+      });
+
+    // metrics
+    const metricUrns = filterPrefix(Object.keys(entities), 'thirdeye:metric:')
+      .filter(urn => urn in scores)
+      .filter(urn => scores[urn] >= ROOTCAUSE_SETUP_METRICS_SCORE_THRESHOLD);
+
+    metricUrns.forEach(urn => newSelectedUrns.add(urn));
```

Comment:
qq: any reason why this wasn't chained after the `.filter`?

Reply:
fixed